### PR TITLE
Fix green flickering when running away

### LINKED ISSUE #18: fix green flickering
when running away they flicker a lot maybe put a minimum time on how long they need to run away.

### PR DESCRIPTION
**Pull request by Billzzhang**
Prey now run a bit after the enemy is out of sight so the enemy doesn't enter sight again.
Uncommon if fov is on

Fix #18

### CODE REVIEW

**Comment by Billzzhang** [comment]
issue  #18

File: src/Helpers/Helpers.js [hunk 1]
```diff
@@ -47,4 +47,9 @@ export function getRandomFloat(max) {
 
 export function round(n, precision) {
   return n.toFixed(precision);
+}
+
+export function getCurrentTime() {
+  let d = new Date;
```
Comment (justinpaoli) [inline src/Helpers/Helpers.js]: should be const

File: src/Creature.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import {getDistance, getNearestDetails, getRandomInt} from "./Helpers/Helpers";
+import {getDistance, getNearestDetails, getRandomInt,getCurrentTime} from "./Helpers/Helpers";
```
Comment (justinpaoli) [inline src/Creature.js]: space after comma

**Comment by devonpmack** [comment]
Good solution

File: src/Constants.js [hunk 3]
```diff
@@ -3,7 +3,8 @@ export const MUTATE_CHANCE = 0.05;
 export const SMALL_MUTATIONS = true;
 export const DEAD_BREED = true; // do creatures that are eaten get to breed
 export const DEAD_BODIES = false; // die after tired dont enable this with disabled dead breed;
-export const USE_FOV = true;
+export const USE_FOV = false;
+export const SCARE_TIME = 500; // time in milliseconds creatures run away when encountering predator
```
Comment (devonpmack) [inline src/Constants.js]: nice 👍

File: src/Creature.js [hunk 4]
```diff
@@ -23,6 +24,8 @@ export default class Creature {
     this.color = [255, 0, 0];
     this.dead = false;
     this.genes = genes;
+    this.scared = false; // in a scared state "retreats" until predator out of sight for SCARE_TIME time
```
Comment (devonpmack) [inline src/Creature.js]: Do we really need this variable? We could just set this.scareTime to null when the object is not scared

File: src/Creature.js [hunk 5]
```diff
@@ -23,6 +24,8 @@ export default class Creature {
...
+    this.scareTime = getCurrentTime();
```
Comment (devonpmack) [inline src/Creature.js]:
```suggestion
    this.scareTime = null;
```

File: src/Creature.js [hunk 6]
```diff
@@ -68,7 +71,7 @@ export default class Creature {
       p5.fill(p5.color(255, 255, 255));
       p5.textSize(fontSize);
       const width = p5.textWidth(this.genes.label);
-      p5.text(this.genes.label, this.x - width / 2, this.y + fontSize / 2)
+      p5.text(this.genes.label, this.x - width / 2, this.y + fontSize / 2);
```
Comment (devonpmack) [inline src/Creature.js]: 🙌

File: src/Creature.js [hunk 7]
```diff
@@ -107,10 +110,18 @@ export default class Creature {
 
     const nearestFood = getNearestDetails(this, this.state.food, true);
 
-    if (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2) {
+    if ((nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2) || this.scared) {
```
Comment (devonpmack) [inline src/Creature.js]:
```suggestion
    if (this.scareTime !== null || (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2)) {
```
This is what I mean by checking for null to see if they are scared. this entire if statement needs refactoring but thats another issue haha.

File: src/Creature.js [hunk 8]
```diff
@@ -107,10 +110,18 @@ export default class Creature {
...
       // run away
       this.color = [0, 255, 0];
-      this.runAwayFrom(nearestCreature.ref);
+      if (!this.scared) {
```
Comment (devonpmack) [inline src/Creature.js]: see above

File: src/Creature.js [hunk 9]
```diff
@@ -107,10 +110,18 @@ export default class Creature {
...
+        this.scared = true;
+        this.scareTime = getCurrentTime(); 
+      } else if ((getCurrentTime() - this.scareTime) > SCARE_TIME ) {
+        if (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2) {
+          this.runAwayFrom(nearestCreature.ref);
+        }
+        this.scared = false;
```
Comment (devonpmack) [inline src/Creature.js]:
```suggestion
        this.scareTime = null;
```

File: src/Creature.js [hunk 10]
```diff
@@ -130,7 +141,7 @@ export default class Creature {
     } else {
       // go towards random position
       this.color = [128, 128, 128];
-      this.moveTowards(this.goingTowards)
+      this.moveTowards(this.goingTowards);
```
Comment (devonpmack) [inline src/Creature.js]: man i really forgot a lot of semi colons when i wrote this 😅

File: src/Helpers/Helpers.js [hunk 11]
```diff
@@ -47,4 +47,9 @@ export function getRandomFloat(max) {
 
 export function round(n, precision) {
   return n.toFixed(precision);
+}
+
+export function getCurrentTime() {
```
Comment (devonpmack) [inline src/Helpers/Helpers.js]: this function is debatable but ill let it be 😉

File: src/Helpers/Helpers.js (hunk 1 above)
Comment (devonpmack) [reply src/Helpers/Helpers.js]: I would actually just not create a variable and return it right away
`return (new Date).getTime()`

**Comment by devonpmack** [comment]
One last thing -- I edited your PR to include `Fix #18` so that it will automatically close #18 when the PR gets merged.

**Comment by devonpmack** [comment]
@Billzzhang can you resolve the things you fixed

File: src/Creature.js [hunk 12]
```diff
@@ -107,10 +109,17 @@ export default class Creature {
 
     const nearestFood = getNearestDetails(this, this.state.food, true);
 
-    if (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2) {
+    if (this.scareTime !== null || (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2)) {
```
Comment (devonpmack) [inline src/Creature.js]: my bad
```suggestion
    if (!this.scaretime || (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2)) {
```

File: src/Creature.js [hunk 13]
```diff
@@ -107,10 +109,17 @@ export default class Creature {
...
       // run away
       this.color = [0, 255, 0];
-      this.runAwayFrom(nearestCreature.ref);
+      if (this.scareTime == null) {
```
Comment (devonpmack) [inline src/Creature.js]: In javascript we want to use triple === @justinpaoli wheres our linting for this
But in this case we can just replace with this for easier reading
```suggestion
      if (this.scareTime) {
```

File: src/Creature.js [hunk 14]
```diff
@@ -107,10 +109,17 @@ export default class Creature {
...
+        this.scareTime = getCurrentTime(); 
+      } else if ((getCurrentTime() - this.scareTime) > SCARE_TIME) {
+        if (nearestCreature && nearestCreature.ref.canEat(this) && nearestCreature.distance < this.size * 2) {
+          this.runAwayFrom(nearestCreature.ref);
+        }
+        this.scareTime = true;
```
Comment (devonpmack) [inline src/Creature.js]: why are we setting it to true?

Comment (Billzzhang) [reply src/Creature.js]: oops, should be null

File: src/Creature.js (hunk 13 above)
Comment (justinpaoli) [reply src/Creature.js]: the linting is there just checked

**Pull request merged**